osu.Game.Rulesets.Osu/UI/ReplayAnalysisOverlay.cs

```diff
@@ -75,7 +83,12 @@ protected override void Update()
         private void initialise()
         {
             if (loaded.IsValid)
+            {
+                if (CursorPath != null)
```

Comment:
What is this conditional doing?

Reply:
If the analysis containers are loaded, check for new frames to add entries for

Reply:
...why is that being achieved via checking `CursorPath`? without any inline commentary? this is so opaque.

whatever this is doing, it needs to be done better. preferably in a way that does not require explanation. even with your explanation above i can barely make out what that is supposed to be doing.

Reply:
Sorry, I should've clarified better. It was to make sure the container variables weren't null (just checking one would be enough), though I just realized if loaded is valid then the containers are also not null.